### PR TITLE
Update to 1.20

### PR DESCRIPTION
**Pull request by Lisixo**
Updated to Minecraft 1.20

I commented [gui/SettingsScreen.java](https://github.com/LambdAurora/LambDynamicLights/compare/1.19.4...Lisixo:LambDynamicLights:1.20#diff-5cbc209c5a3771f09609a8b63ad883dda29f833e836a5a56420448410b34072a) because i can't find solution for this code fragment but mod and game work fine without this code fragment

### CODE REVIEW

**Review by Kichura — COMMENTED** [review]
Tested this under a clean fabric instance and it seems to work just fine, not sure what the `fillGradient` in settings screen system was for though.

**Comment by Lisixo** [comment]
`fillGradient` adds black shadow to settings background

Latest 1.19.4 build:
ss_1 19 4

1.20 build:
ss_1 20

File: src/main/java/dev/lambdaurora/lambdynlights/gui/SettingsScreen.java [hunk 1]
```diff
@@ -136,12 +135,14 @@ private SpruceContainerWidget buildTabContainer(int width, int height, SpruceTab
...
-						widget.getX() + widget.getWidth(), widget.getY() + widget.getHeight(),
-						0xc0101010, 0xd0101010);
+				// Commented because i can't find a solution
+
+//				this.fillGradient(matrices, widget.getX(), widget.getY(),
+//						widget.getX() + widget.getWidth(), innerWidget.getY(),
+//						0xc0101010, 0xd0101010);
+//				this.fillGradient(matrices, widget.getX(), innerWidget.getY() + innerWidget.getHeight(),
+//						widget.getX() + widget.getWidth(), widget.getY() + widget.getHeight(),
+//						0xc0101010, 0xd0101010);
```
Comment (Oliver-makes-code) [inline src/main/java/dev/lambdaurora/lambdynlights/gui/SettingsScreen.java]:
```suggestion
		container.setBackground((graphics, widget, vOffset, mouseX, mouseY, delta) -> {
			if (this.client.world != null) {
				graphics.fillGradient(widget.getX(), widget.getY(),
						widget.getX() + widget.getWidth(), innerWidget.getY(),
						0xc0101010, 0xd0101010);
				graphics.fillGradient(widget.getX(), innerWidget.getY() + innerWidget.getHeight(),
						widget.getX() + widget.getWidth(), widget.getY() + widget.getHeight(),
						0xc0101010, 0xd0101010);
```
This should be what you need

**Review by Oliver-makes-code — CHANGES_REQUESTED** [review]
Mojang doesn't pass MatixStacks in gui anymore, they use a class called GuiGraphics, which contains the stuff needed.

File: src/main/java/dev/lambdaurora/lambdynlights/gui/SettingsScreen.java (hunk 1 above)
Comment (Lisixo) [reply src/main/java/dev/lambdaurora/lambdynlights/gui/SettingsScreen.java]: Thank you. Now everything should be fine

**Comment by AgentKerbal** [comment]
Hi, bit of a noob problem that probably has nothing to do with your pull, but I can't get it to build with gradle, it fails on "checkLicenseMain", any idea ?

**Comment by jhmaster2000** [comment]
If anyone came here looking for a build, here you go:
https://github.com/jhmaster2000/LambDynamicLights/actions/runs/5239043539 (Scroll down to **Artifacts**)

I simply forked Lisixo's fork and enabled the official build workflow and ran it as-is, so you can know it's a legit build.

**Comment by Rubyboat1207** [comment]
This works in 1.20.1, someone needs to update the mod.json though

**Comment by joestr** [comment]
Confirmed working on 1.20.1

**Comment by mk-pmb** [comment]
Can confirm, works for me for 1.20. [This is my build](https://github.com/mk-pmb/ghabuild-simple-gradle-projects/releases/tag/rolling/gh/lambdaurora/lambdynamiclights) including a JAR download for anyone else who needs one.

@Lisixo Thanks a lot!

**Comment by PetyXbron** [comment]
Hey @Lisixo, could this PR also support Minecraft v1.20.1?
image

**Comment by mk-pmb** [comment]
Thanks @KennyDevT ! I restarted my build. The new JAR for 1.20.1 should be public in about 10 minutes.
Edit: The new download has arrived.

**Comment by KennyDevT** [comment]
Honestly I just wanted to have it in my modpack I didn't even know about this PR until I decided to PR it to Lixiso fork

**Comment by JanLukasGithub** [comment]
Now we just need @LambdAurora (or someone else with rights to do so) to merge this and update the modrinth page, right?

**Comment by mk-pmb** [comment]
@JanLukasGithub I think we should wait for more people to report whether it worked for them in 1.20.1. I myself only play 1.20 currently, and I haven't seen any confirmation comments for the current commit https://github.com/LambdAurora/LambDynamicLights/pull/156/commits/fea5d26f2179cad66cc452393314326b68864f75 .

Update: [fea5d26 JAR download](https://github.com/mk-pmb/ghabuild-simple-gradle-projects/releases/download/rolling%2Fgh%2Flambdaurora%2Flambdynamiclights/lambdynamiclights-v2.3.1-mc1.20.1-230619-003049.jar) if someone needs it. React with 🚀 if it works for you __in MC 1.20.1__.

**Comment by AuroraDoesCode** [comment]
I know this is closed and published now, but Fabric API should be 0.83.0 if people are still using base 1.20, as most mod launchers will still serve it as the most recent version for 1.20.

**Comment by LambdAurora** [comment]
Well, that's one issue I just realized I forgot.
The dependency should also be marked `fabric-api` now... oops...

Welp, I'm not fixing that now, worst case scenario there's dependency overrides, and 1.20.1 is so minor everything should work there anyway so people should just update.

**Comment by MBCMechachu** [comment]
That's a tad unfortunate. I was kind of hoping to use the mod in 1.20.0 but each time I loaded it in with my other mods I kept getting this message.

Replace mod 'Fabric API' (fabric-api) 0.83.0+1.20 with mod 'Fabric API' (fabric-api), version 0.83.1+1.20.1 or later.
<html>Mod 'LambDynamicLights' (lambdynlights) 2.3.1+1.20.1 requires version 0.83.1+1.20.1 or later of mod 'Fabric API'<br>(fabric-api), but only the wrong version is present: 0.83.0+1.20!</html>

**Comment by LambdAurora** [comment]
@MBCMechachu update to 1.20.1 and update Fabric API.

1.20.1 is so minor that all mods should work and be compatible with 1.20 servers.

**Comment by AuroraDoesCode** [comment]
> That's a tad unfortunate. I was kind of hoping to use the mod in 1.20.0 but each time I loaded it in with my other mods I kept getting this message.
> 
> Replace mod 'Fabric API' (fabric-api) 0.83.0+1.20 with mod 'Fabric API' (fabric-api), version 0.83.1+1.20.1 or later.
> 
> Mod 'LambDynamicLights' (lambdynlights) 2.3.1+1.20.1 requires version 0.83.1+1.20.1 or later of mod 'Fabric API'
> (fabric-api), but only the wrong version is present: 0.83.0+1.20!

You can create an overrides pack to bypass that, follow https://fabricmc.net/wiki/tutorial:dependency_overrides and just set it up for dynamic lights, or update to 1.20.1, and just copy paste everything, as it should work anyway.

**Comment by MBCMechachu** [comment]
> > That's a tad unfortunate. I was kind of hoping to use the mod in 1.20.0 but each time I loaded it in with my other mods I kept getting this message.
> > Replace mod 'Fabric API' (fabric-api) 0.83.0+1.20 with mod 'Fabric API' (fabric-api), version 0.83.1+1.20.1 or later.
> > Mod 'LambDynamicLights' (lambdynlights) 2.3.1+1.20.1 requires version 0.83.1+1.20.1 or later of mod 'Fabric API'
> > (fabric-api), but only the wrong version is present: 0.83.0+1.20!
> 
> You can create an overrides pack to bypass that, follow https://fabricmc.net/wiki/tutorial:dependency_overrides and just set it up for dynamic lights, or update to 1.20.1, and just copy paste everything, as it should work anyway.

A lot of this is far too complex for me to understand in order for me to do this on my own. What's worse is that I can't seem to find any sort of step by step videos anywhere so I'm just stuck having to wait until the mods I was using update to 1.20.1; assuming that some of them do at all.

Update: Turns out all I needed to do was update my Fabric API to the one for 1.20.1, everything is working fine now.

**Comment by MBCMechachu** [comment]
> @MBCMechachu update to 1.20.1 and update Fabric API.
> 
> 1.20.1 is so minor that all mods should work and be compatible with 1.20 servers.

As simple as that would be a bunch of the mods I would've had running in tandem such as MouseWheelie aren't updated to that version of Minecraft at the time I'm replying back to you.

**Comment by mk-pmb** [comment]
Can we have [the v2.3.1+1.20.1 release page](https://github.com/LambdAurora/LambDynamicLights/releases/tag/v2.3.1%2B1.20.1) have a big hint "You'll need to update Fabric API to at least x.x.x" also with a download link to that?

Edit: Maybe it would be a good idea to have a default "System Requirements" chapter as one of the first things on every release page.

**Pull request merged**